src/binder/bind/bind_copy.cpp

```diff
@@ -21,32 +21,32 @@ std::unique_ptr<BoundStatement> Binder::bindCopyToClause(const Statement& statem
     auto boundFilePath = copyToStatement.getFilePath();
     auto fileType = bindFileType(boundFilePath);
     std::vector<std::string> columnNames;
-    std::vector<LogicalType> columnTypes;
+    std::vector<std::unique_ptr<LogicalType>> columnTypes;
     auto query = bindQuery(*copyToStatement.getRegularQuery());
     auto columns = query->getStatementResult()->getColumns();
     for (auto& column : columns) {
         auto columnName = column->hasAlias() ? column->getAlias() : column->toString();
         columnNames.push_back(columnName);
-        columnTypes.push_back(column->getDataType());
+        columnTypes.push_back(column->getDataType().copy());
     }
-    if (fileType != CopyDescription::FileType::CSV &&
-        fileType != CopyDescription::FileType::PARQUET) {
+    if (fileType != FileType::CSV && fileType != FileType::PARQUET) {
         throw BinderException(ExceptionMessage::validateCopyToCSVParquetExtensionsException());
     }
-    auto copyDescription = std::make_unique<CopyDescription>(
-        fileType, std::vector<std::string>{boundFilePath}, nullptr /* parsing option */);
-    copyDescription->columnNames = columnNames;
-    copyDescription->columnTypes = columnTypes;
-    return std::make_unique<BoundCopyTo>(std::move(copyDescription), std::move(query));
+    auto readerConfig = std::make_unique<ReaderConfig>();
```

Comment:
Can we create a new constructor for ReaderConfig which takes in the fileType, filePaths, columnNames, columnTypes? So you don't need to set each field manually.